tests/native_tests/test_metaclass.hy

```diff
@@ -0,0 +1,15 @@
+;; Copyright 2018 the authors.
+;; This file is part of Hy, which is free software licensed under the Expat
+;; license. See the LICENSE.
+
+(defn test-metaclass []
+  (defclass QuestBase [type]
+     [--new-- (fn [cls name bases classdict]
+               (if (in "__swallow__" classdict)
+                   (assoc classdict "swallow" (get classdict "__swallow__")))
+               (-> (super QuestBase cls) (.--new-- cls name bases classdict)))])
+
+  (defclass Quest [:metaclass QuestBase]
+    [--swallow-- "african"])
```

Comment:
Why the use of double underscores? `__swallow__` isn't a special Python name, which is what double underscores on both sides of a name are typically reserved for, and it doesn't trigger the automatic Python mangling that a name like `__swallow` would, which we probably don't need for this test, anyway.

Reply:
If all you want to do is distinguish `swallow` from a similar but distinct variable, how about `swallow_`?

Reply:
No good reason. Honestly, I'm not married to this particular test case at all. The other two came from their respective PEP's, this one I derived.